### ISSUE TITLE
Introduce "use implicit array creation" analyzer

### ISSUE THREAD

**Issue by mrahhal** [dotnet/roslyn#57788]
Introduce a "use implicit array creation" analyzer that transforms an `ArrayCreationExpression` to an `ImplicitArrayCreationExpression` where applicable.

**Languages applicable:**

C# only.

**Code examples:**

1. Basic case:
```csharp
var a = new [|int|][] { 1, 2 };
```

2. Shouldn't report when removing the type changes the semantics of the creation expression. Example:
```csharp
// Removing the type here will result in changing the numbers array into int[] instead, so don't report this.
var numbers = new double[] { 1, 2 };
```

3. In case of this:
```csharp
var a = new C[] { new C(), new C() };
````
I think the inner implicit object creation analyzer should take precedence after #57789 is done.

**Documentation requirements:**

Will require an addition to the docs at:  https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md.

I would like to work on this analyzer if this is deemed a good addition.

**Comment by CyrusNajmabadi** [comment]
Will bring to next design meeting.  However, i think we would definitely take this.

**Comment by CyrusNajmabadi** [comment]
@sharwell I'd like to bring this to next meeting.

**Comment by CyrusNajmabadi** [comment]
Closing out as 'use collection expression' fixer subsumes this case.
